### PR TITLE
feat(batch): add Batch labels allocation sample

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by Sita04** [comment]
Please follow https://www.conventionalcommits.org/en/v1.0.0/ for PR naming.

File: batch/snippets/src/main/java/com/example/batch/CreateBatchAllocationPolicyLabel.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    String labelValue2 = "VM_LABEL_VALUE2";
+
+    createBatchAllocationPolicyLabel(projectId, region, jobName, labelName1,
+        labelValue1, labelName2, labelValue2);
+  }
+
+  //This method shows how to create the job with labels defined in the labels field
+  // for a job's allocation policy, as well as to each GPU (if any),
+  // persistent disk (all boot disks and any new storage volumes),
+  // and VM created for the job.
```
Comment (Sita04) [inline batch/snippets/src/main/java/com/example/batch/CreateBatchAllocationPolicyLabel.java:63]:
```suggestion
  // This method shows how to create a job with labels defined 
  // in the labels field of a job's allocation policy. These are 
  // applied to the job, as well as to each GPU (if any), persistent disk 
  // (all boot disks and any new storage volumes), and VM created for the job.
```

File: batch/snippets/src/main/java/com/example/batch/CreateBatchAllocationPolicyLabel.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+          AllocationPolicy.InstancePolicy.newBuilder().setMachineType("e2-standard-4").build();
+
+      AllocationPolicy allocationPolicy =
+          AllocationPolicy.newBuilder()
+              .addInstances(AllocationPolicy.InstancePolicyOrTemplate.newBuilder()
+                  .setPolicy(instancePolicy).build())
+                  // Labels and their value to be applied to the job and its resources
+                  .putLabels(labelName1, labelValue1)
+                  .putLabels(labelName2, labelValue2)
+                  .build();
```
Comment (Sita04) [inline batch/snippets/src/main/java/com/example/batch/CreateBatchAllocationPolicyLabel.java:121]:
```suggestion
              .addInstances(AllocationPolicy.InstancePolicyOrTemplate.newBuilder()
                  .setPolicy(instancePolicy)
                  .build())
              // Labels and their value to be applied to the job and its resources
              .putLabels(labelName1, labelValue1)
              .putLabels(labelName2, labelValue2)
              .build();
```
The labels are applied to the allocation policy and should be intended to denote that.

File: batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java [hunk 3]
```diff
@@ -58,6 +58,8 @@ public class CreateResourcesIT {
           + UUID.randomUUID().toString().substring(0, 7);
   private static final String CUSTOM_EVENT_NAME = "test-job"
           + UUID.randomUUID().toString().substring(0, 7);
+  private static final String JOB_ALLOCATION_POLICY_LABEL = "test-job"
```
Comment (Sita04) [inline batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java:61]:
```suggestion
  private static final String JOB_ALLOCATION_POLICY_LABEL = "test-job-allocation-label"
```

Always have a unique suffix to every job name to denote the job.

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java [hunk 4]
```diff
@@ -59,7 +59,11 @@ public class CreateResourcesIT {
           + UUID.randomUUID().toString().substring(0, 7);
   private static final String CUSTOM_EVENT_NAME = "test-job"
           + UUID.randomUUID().toString().substring(0, 7);
-  private static final String JOB_ALLOCATION_POLICY_LABEL = "test-job"
+  private static final String BATCH_LABEL_JOB = "test-job-label"
+      + UUID.randomUUID().toString().substring(0, 7);
+  private static final String CUSTOM_NETWORK_NAME = "test-job-network"
+      + UUID.randomUUID().toString().substring(0, 7);
```
Comment (Sita04) [inline batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java:65]: Remove this.

File: batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java [hunk 5]
```diff
@@ -251,6 +258,68 @@ public void createBatchCustomEventTest()
                     .anyMatch(runnable -> runnable.getDisplayName().equals(displayName))));
   }
 
+  @Test
```
Comment (Sita04) [inline batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java:261]: same comment as in the other PR. Kindly look at the changes unrelated to this PR and remove them.

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java (hunk 4 above)
Comment (TetyanaYahodska) [reply batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java:65]: This code is in the master branch that was merged earlier. Two samples are tested using these variables. Why do I need to remove this code?

File: batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java (hunk 5 above)
Comment (TetyanaYahodska) [reply batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java]: In this PR I added only one test for Batch allocation labels. The last commit was  "Merged changes from main, resolved merge conflict", to keep my branch updated. Please review my code once more.

File: batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java (hunk 4 above)
Comment (TetyanaYahodska) [reply batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java]: I think this is a misunderstanding, please review my PR again. I don't have this piece of code as a new added. On the "Files Changed" page, I only see the code related to this PR.

Comment (Sita04) [reply batch/snippets/src/test/java/com/example/batch/CreateResourcesIT.java]: I think it was a dirty merge in the last commit. The latest one has  pulled in the changes.

**Review by Sita04 — APPROVED** [review]

**Pull request merged**